tests/functional/test_fusion_hap.py

```diff
@@ -0,0 +1,1024 @@
+# -*- coding: utf-8 -*-
+
+# (c) 2023, Andrej Pajtas (apajtas@purestorage.com)
+# GNU General Public License v3.0+ (see COPYING.GPLv3 or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+from unittest.mock import MagicMock, patch
+
+import fusion
+import fusion as purefusion
+import pytest
+from ansible.module_utils import basic
+from ansible_collections.purestorage.fusion.plugins.module_utils.errors import (
+    OperationException,
+)
+from ansible_collections.purestorage.fusion.plugins.modules import fusion_hap
+from ansible_collections.purestorage.fusion.tests.functional.utils import (
+    AnsibleExitJson,
+    AnsibleFailJson,
+    FailedOperationMock,
+    OperationMock,
+    SuccessfulOperationMock,
+    exit_json,
+    fail_json,
+    set_module_args,
+)
+from urllib3.exceptions import HTTPError
+
+# GLOBAL MOCKS
+fusion_hap.setup_fusion = MagicMock(return_value=purefusion.api_client.ApiClient())
+purefusion.api_client.ApiClient.call_api = MagicMock(
+    side_effect=Exception("API call not mocked!")
+)
+basic.AnsibleModule.exit_json = exit_json
+basic.AnsibleModule.fail_json = fail_json
+
+
+@pytest.fixture
+def module_args():
+    """Module arguments to create new HAP."""
+    return {
+        "state": "present",
+        "name": "hap_new",
+        "display_name": "Host Access Policy New",
+        "nqn": "string_value",
+        "iqn": "iqn.2023-05.com.purestorage:420qp2c0699",
+        "wwns": ["wwn1", "wwn2"],
+        "host_password": "u^nv2320XnYQ",
```

Comment:
I didn't notice before that Ansible HAP has these. We should remove them, they are not supported and will just cause confusion. Lets do it in a separate task.

Reply:
Agree.